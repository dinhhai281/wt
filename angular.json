{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "projects",
  "projects": {
    "angular-unit-test": {
      "projectType": "application",
      "schematics": {
        "@schematics/angular:component": {
          "style": "scss",
          "changeDetection": "OnPush",
          "viewEncapsulation": "None"
        }
      },
      "root": "",
      "sourceRoot": "src",
      "prefix": "app",
      "architect": {
        "build": {
          "builder": "@angular-builders/custom-webpack:browser",
          "options": {
            "outputPath": "dist/angular-unit-test",
            "index": "src/index.html",
            "main": "src/main.ts",
            "polyfills": "src/polyfills.ts",
            "tsConfig": "tsconfig.app.json",
            "aot": true,
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ],
            "styles": [
              "src/styles.scss"
            ],
            "scripts": [],
            "customWebpackConfig": {
              "path": "webpack.config.js"
            }
          },
          "configurations": {
            "production": {
              "fileReplacements": [
                {
                  "replace": "src/environments/environment.ts",
                  "with": "src/environments/environment.prod.ts"
                }
              ],
              "optimization": true,
              "outputHashing": "all",
              "sourceMap": false,
              "extractCss": true,
              "namedChunks": false,
              "extractLicenses": true,
              "vendorChunk": false,
              "buildOptimizer": true,
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "2mb",
                  "maximumError": "5mb"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "6kb",
                  "maximumError": "10kb"
                }
              ]
            }
          }
        },
        "serve": {
          "builder": "@angular-builders/custom-webpack:dev-server",
          "options": {
            "browserTarget": "angular-unit-test:build",
            "customWebpackConfig": {
              "path": "webpack.config.js"
            }
          },
          "configurations": {
            "production": {
              "browserTarget": "angular-unit-test:build:production"
            }
          }
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "browserTarget": "angular-unit-test:build"
          }
        },
        "lint": {
          "builder": "@angular-eslint/builder:lint",
          "options": {
            "lintFilePatterns": [
              "src/**/*.ts",
              "src/**/*.component.html"
            ],
            "exclude": [
              "**/node_modules/**"
            ]
          }
        }
      }
    }
  },
  "defaultProject": "angular-unit-test"
}
