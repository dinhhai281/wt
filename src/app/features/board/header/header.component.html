<ng-container *ngIf="state.vm$ | async as vm">
  <div class="flex py-3 px-8 justify-between items-center bg-gray-800 shadow-sm">
    <section class="flex items-center">
      <div
        class="h-8 w-8 bg-white text-blue-500 rounded-full border-2 border-gray-400 flex items-center justify-center text-xs"
      >
        <ng-content select="[app-label]"></ng-content>
      </div>
      <input
        placeholder="Search or jump to..."
        class="ml-4 h-6 px-2 text-sm rounded bg-gray-900 text-gray-200 focus:outline-none focus:shadow-outline"
      />
    </section>
    <section class="flex justify-center items-center">
      <mat-icon
        class="cursor-pointer mr-4"
        matBadge="3"
        matBadgeColor="primary"
        matBadgeSize="small"
        svgIcon="notification"
      ></mat-icon>
      <button class="mr-4 focus:outline-none flex items-center">
        <mat-icon svgIcon="add"></mat-icon>
      </button>
      <button class="focus:outline-none flex items-center"
        (click)="state.toggleUserMenu()"
        cdkOverlayOrigin
        #userMenu="cdkOverlayOrigin"
      >
        <span
          class="h-8 w-8 rounded-full text-blue-500 bg-white border-2 border-gray-400 flex items-center justify-center text-xs"
        >
          {{ vm.me | userAvatar:'NA' }}
        </span>
      </button>
    </section>
  </div>
  <ng-template
    cdkConnectedOverlay
    [cdkConnectedOverlayOrigin]="userMenu"
    [cdkConnectedOverlayOpen]="vm.isOpenUserMenu"
    [cdkConnectedOverlayOffsetY]="8"
    (overlayOutsideClick)="$event.stopImmediatePropagation(); state.toggleUserMenu();"
  >
    <section class="flex flex-col rounded-lg bg-gray-800 w-48 border border-gray-900">
      <div class="p-3 text-gray-200 flex flex-col border-b border-gray-900">
        <span>Signed in as</span>
        <span class="font-semibold text-gray-100">{{ 'NA' }}</span>
      </div>

      <ul class="py-3 text-gray-200 flex flex-col">
        <li class="px-3 py-1 cursor-pointer hover:bg-blue-400">Your profile</li>
        <li class="px-3 py-1 cursor-pointer hover:bg-blue-400">Your repositories</li>
        <li class="px-3 py-1 cursor-pointer hover:bg-blue-400">Your organizations</li>
        <li class="px-3 py-1 cursor-pointer hover:bg-blue-400">Your projects</li>
        <li class="px-3 py-1 cursor-pointer hover:bg-blue-400">Your starts</li>
        <li class="px-3 py-1 cursor-pointer hover:bg-blue-400">Your gists</li>
      </ul>
    </section>
  </ng-template>
</ng-container>


