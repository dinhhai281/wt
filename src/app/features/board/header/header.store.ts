import { Injectable } from '@angular/core';
import { User } from '@app/models';
import { ComponentStore } from '@ngrx/component-store';

export interface HeaderState {
  me: User;
  isOpenUserMenu: boolean;
}

@Injectable()
export class HeaderStore extends ComponentStore<HeaderState> {
  constructor() {
    super({ me: null, isOpenUserMenu: false });
  }

  readonly me$ = this.select(state => state.me);
  readonly isOpenUserMenu$ = this.select(state => state.isOpenUserMenu);
  readonly vm$ = this.select(this.me$, this.isOpenUserMenu$, (me, isOpenUserMenu) => ({
    me,
    isOpenUserMenu,
  }));

  readonly setMe = this.updater<User>((state, me) => ({
    ...state,
    me,
  }));
  readonly toggleUserMenu = this.updater<boolean>(state => ({
    ...state,
    isOpenUserMenu: !state.isOpenUserMenu,
  }));
}
