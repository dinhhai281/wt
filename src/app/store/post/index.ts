export * from './post.action';
export * from './post.reducer';
export * from './post.effect';
export * from './post.selector';
export * from './post.module';
