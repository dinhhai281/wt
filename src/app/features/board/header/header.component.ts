import {
  Component,
  OnInit,
  ViewEncapsulation,
  ChangeDetectionStrategy,
  Input,
} from '@angular/core';
import { User } from '@app/models';
import { HeaderStore } from './header.store';

@Component({
  selector: 'app-header',
  templateUrl: './header.component.html',
  styleUrls: ['./header.component.scss'],
  encapsulation: ViewEncapsulation.None,
  changeDetection: ChangeDetectionStrategy.OnPush,
  providers: [HeaderStore],
})
export class HeaderComponent implements OnInit {
  @Input()
  set me(value: User) {
    this.state.setMe(value);
  }

  ngOnInit() {}

  constructor(public readonly state: HeaderStore) {}
}
