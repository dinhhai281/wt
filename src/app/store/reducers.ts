import { ActionReducerMap } from '@ngrx/store';
import * as fromPost from '@store/entities/post';
import * as fromAuth from '@store/application/auth';

export interface State {
  [fromPost.POST_FEATURE_KEY]: fromPost.PostEntityState;
  [fromAuth.AUTH_FEATURE_KEY]: fromAuth.AuthState;
}

export { fromPost, fromAuth };
