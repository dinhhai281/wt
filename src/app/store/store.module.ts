import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { environment } from '@environments/environment';
import { EffectsModule } from '@ngrx/effects';
import { StoreModule } from '@ngrx/store';

import { AuthFeatureStoreModule } from './application/auth';
import { PostFeatureStoreModule } from './entities/post';

@NgModule({
  declarations: [],
  imports: [
    CommonModule,
    StoreModule.forRoot(
      {},
      {
        runtimeChecks: {
          strictActionImmutability: true,
          strictStateSerializability: true,
          strictActionSerializability: true,
          strictStateImmutability: true,
          strictActionTypeUniqueness: true,
        },
      },
    ),
    EffectsModule.forRoot(),
    environment.storeInstrument,
    PostFeatureStoreModule,
    AuthFeatureStoreModule,
  ],
})
export class AppStoreModule {}
