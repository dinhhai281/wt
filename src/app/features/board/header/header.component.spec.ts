import { OverlayModule } from '@angular/cdk/overlay';
import { MatIconModule, MatIconRegistry } from '@angular/material/icon';
import { FakeMatIconRegistry } from '@angular/material/icon/testing';
import { render, RenderResult } from '@testing-library/angular';

import { UserAvatarPipe } from '../pipes/user-avatar.pipe';
import { HeaderComponent } from './header.component';
import { HeaderStore } from './header.store';

describe('HeaderComponent', () => {
  let component: RenderResult<HeaderComponent>;

  beforeEach(async () => {
    component = await render(HeaderComponent, {
      declarations: [UserAvatarPipe],
      imports: [MatIconModule, OverlayModule],
      providers: [{ provide: MatIconRegistry, useClass: FakeMatIconRegistry }, HeaderStore],
    });
  });

  it('should match snapshot', () => {
    expect(component.container).toMatchSnapshot();
  });
});
