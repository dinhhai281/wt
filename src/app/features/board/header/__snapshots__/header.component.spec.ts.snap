// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent should match snapshot 1`] = `
<div
  id="root0"
>
  
  
  <div
    class="flex py-3 px-8 justify-between items-center bg-gray-800 shadow-sm"
  >
    <section
      class="flex items-center"
    >
      <div
        class="h-8 w-8 bg-white text-blue-500 rounded-full border-2 border-gray-400 flex items-center justify-center text-xs"
      />
      <input
        class="ml-4 h-6 px-2 text-sm rounded bg-gray-900 text-gray-200 focus:outline-none focus:shadow-outline"
        placeholder="Search or jump to..."
      />
    </section>
    <section
      class="flex justify-center items-center"
    >
      <mat-icon
        aria-hidden="true"
        class="cursor-pointer mr-4 mat-icon notranslate mat-icon-no-color"
        data-mat-icon-name="notification"
        data-mat-icon-type="svg"
        matbadge="3"
        matbadgecolor="primary"
        matbadgesize="small"
        role="img"
        svgicon="notification"
      >
        <svg
          class="fake-testing-svg"
          fit=""
          focusable="false"
          height="100%"
          preserveAspectRatio="xMidYMid meet"
          width="100%"
        />
      </mat-icon>
      <button
        class="mr-4 focus:outline-none flex items-center"
      >
        <mat-icon
          aria-hidden="true"
          class="mat-icon notranslate mat-icon-no-color"
          data-mat-icon-name="add"
          data-mat-icon-type="svg"
          role="img"
          svgicon="add"
        >
          <svg
            class="fake-testing-svg"
            fit=""
            focusable="false"
            height="100%"
            preserveAspectRatio="xMidYMid meet"
            width="100%"
          />
        </mat-icon>
      </button>
      <button
        cdkoverlayorigin=""
        class="focus:outline-none flex items-center"
      >
        <span
          class="h-8 w-8 rounded-full text-blue-500 bg-white border-2 border-gray-400 flex items-center justify-center text-xs"
        >
           NA 
        </span>
      </button>
    </section>
  </div>
  
  
</div>
`;
